 your communities  Find centralized, trusted content and collaborate around the technologies you use most.  Connect and share knowledge within a single location that is structured and easy to search.  I have a POJO class with one of array property having size is 2. But I am creating the object of class with passing array of size 5. It's not showing any exception. Why?  When you define/initialize the array correctly the compiler won't complain and no error occurs. However when you run your program with e.g. size more than what your array allows you are going to get a such as will be thrown when trying to get an item with index e.g. in your case.  2 Answers 2  is an object, although it doesn't look like it. So when you declare.  You're actually declaring a pointer to a new object, which is a object, with a size of .  Then you come a long with a new object of size , and your reference now points to that.  This process has nothing to do with the original object, because when you state:  you're not effecting the object that points to; only the pointer itself. The object will have no pointers referring to it, and the garbage collector will likely come a long and destroy it.  You're pointing the class variable to the argument passed in the function... Check its' size in the constructor and if it's not 2, then throw IllegalArgumentExepction.  By clicking “Accept all cookies”, you agree Stack Exchange can store cookies on your device and disclose information in accordance with our Cookie Policy.  